src/ConstructiveSolidGeometry/LinePrimitives/Line.jl

```diff
@@ -0,0 +1,74 @@
+struct Line{T,TP,TL} <: AbstractLinePrimitive{T}
+    p1::TP
+    p2::TP
+    linetype::TL
+    function Line( ::Type{T},
+                   p1::Union{CartesianPoint{T}, PlanarPoint{T}},
+                   p2::Union{CartesianPoint{T}, PlanarPoint{T}},
+                   linetype::Union{Val{:inf}, Val{:ray}, Val{:seg}}) where {T}
+        new{T,typeof(p1),typeof(linetype)}(p1, p2)
+    end
+end
+
+function Line( ::Type{T},
+               p1::Union{CartesianPoint{T}, PlanarPoint{T}},
+               p2::Union{CartesianPoint{T}, PlanarPoint{T}}) where {T} #if (inf)Line there are no endpoints
+    Line(T, p1, p2, Val(:inf))
+end
+
+function Ray( ::Type{T},
+               p1::Union{CartesianPoint{T}, PlanarPoint{T}},
+               p2::Union{CartesianPoint{T}, PlanarPoint{T}}) where {T} #if Ray endpoint is p1
+    Line(T, p1, p2, Val(:ray))
+end
+
+function LineSegment( ::Type{T},
+               p1::Union{CartesianPoint{T}, PlanarPoint{T}},
+               p2::Union{CartesianPoint{T}, PlanarPoint{T}}) where {T} #if LineSegment endpoints are p1, and p2
+    Line(T, p1, p2, Val(:seg))
+end
+
+Line(l::Line{T}) where {T} = Line(T, l.p1, l.p2)
+
+get_line_vector(l::Line{T,PlanarPoint{T}}) where {T} = normalize(PlanarVector{T}(l.p2 - l.p1))
+get_line_vector(l::Line{T,CartesianPoint{T}}) where {T} = normalize(CartesianVector{T}(l.p2 - l.p1))
```

Comment:
We could override `Base.-` such that a difference of two `Points{T}` gives a `Vector{T}`.
Then this could be summarized to `get_line_vector(l::Line) = normalize(l.p2 - l.p1)`?
Also: If this is needed often, does it make sense to save it to the struct `Line{T}`?

Reply:
Alright, then this remains as is!